### PR TITLE
feat: Refactor version anchor

### PR DESCRIPTION
**Pull request by wzalazar**
[PR Process](https://github.com/poetapp/documentation/blob/master/process/pr-review.md#pr-process) - [PR Review Checklist](https://github.com/poetapp/documentation/blob/master/process/pr-review.md#pr-review-checklist)

### Release
 Semantic release is enabled for this repository. Make sure you follow the right commit message convention. 
 We're using semantic-release's default — [Angular Commit Message Conventions](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines).

## Description of Changes

Fix https://github.com/poetapp/node/issues/491

This PR introduces:

- Read and write the version anchor like UInt16BE

### CODE REVIEW

File: src/Configuration.ts [hunk 1]
```diff
@@ -232,14 +232,10 @@ function loadConfigurationFromEnv(env: any): Partial<Configuration> {
...
-  assert(Array.isArray(poetVersion), 'Field poetVersion must be an Array')
-  assert(poetVersion.length === 4, 'Field poetVersion must have 4 elements')
-  poetVersion.forEach((element: any) =>
-    assert(
-      Number.isInteger(element) && 0 <= element && element < 256,
-      'Each member of poetVersion must be an integer between 0 and 255',
-    ),
+function validatePoetVersion(poetVersion: number) {
+  assert(
+    Number.isInteger(poetVersion) && 0 <= poetVersion && poetVersion < 4294967296,
```
Comment (lautarodragan) [inline src/Configuration.ts]: Shouldn't this be 65536? (256 ** 2)

File: src/BlockchainWriter/Bitcoin.ts [hunk 2]
```diff
@@ -7,13 +7,26 @@ export class IllegalPrefixLength extends Error {}
 
 export class IllegalVersionLength extends Error {}
 
+const uint16ToBuffer = (version: number) => {
+  const buffer = Buffer.alloc(2)
+
+  try {
+    buffer.writeUInt16BE(version, 0)
+  } catch {
```
Comment (lautarodragan) [inline src/BlockchainWriter/Bitcoin.ts]: Why are we replacing any error thrown by `writeUInt16BE` with `IllegalVersionLength` here?

Is an incorrect version length the only thing that could go bad?

File: src/BlockchainReader/Bitcoin.ts [hunk 3]
```diff
@@ -78,7 +78,8 @@ export const isCorrectBufferLength = (buffer: Buffer) => buffer.byteLength >= an
 
 export const bufferToPoetAnchor = (buffer: Buffer): PoetAnchor => {
   const prefix = buffer.slice(0, 4).toString()
-  const version = Array.from(buffer.slice(4, 6))
+  // const version = buffer.slice(4, 6).readUInt16BE(0)
```
Comment (lautarodragan) [inline src/BlockchainReader/Bitcoin.ts]: Left over comment?

File: src/Configuration.ts (hunk 1 above)
Comment (wzalazar) [reply src/Configuration.ts]: I played around with writeUInt16BE and writeUInt16LE and I found that the maximums numbers are:

```
writeUInt16LE 65535 (2 ^ 16)
writeUInt16BE 4294967296 (2 ^ 32)
```

File: src/BlockchainReader/Bitcoin.ts (hunk 3 above)
Comment (wzalazar) [reply src/BlockchainReader/Bitcoin.ts]: good catch!

File: src/Configuration.ts (hunk 1 above)
Comment (lautarodragan) [reply src/Configuration.ts]: I'm confused. How can endianness affect the size?

Comment (wzalazar) [reply src/Configuration.ts]: well, I received this error when you are out the range:

writeUInt16BE:

`RangeError [ERR_OUT_OF_RANGE]: The value of "value" is out of range. It must be >= 0 and <= 4294967295. Received 4294967296`

writeUInt16LE:
`RangeError [ERR_OUT_OF_RANGE]: The value of "value" is out of range. It must be >= 0 and <= 65535. Received -1`

Comment (lautarodragan) [reply src/Configuration.ts]: Raised an issue about this to NodeJS and they already fixed it! :smile: 

https://github.com/nodejs/node/issues/24205

So we should update our code accordingly.

`Number.isInteger(poetVersion) && 0 <= poetVersion && poetVersion <= 0xFFFF`

**Pull request merged**